gnosis/safe-react
# fix: cache queue

## What it solves
Resolves #3704

## How this PR fixes it
The transaction queue is now cached and previous deeplink logic has been modified:

- Adding queued transactions first checks for the existence of a transaction to add by `txId`. 'New' transactions overwrite the cache if more confirmations have been submitted since being cache.
- The Safe no longer re-requests the queue if only one transaction is present in the store, e.g. a singular transaction was loaded. This was the root cause of the bug.
- Singular transaction view (deeplink) first checks if the requested transaction exists in the store. As we fetch the queued transactions in the `TxSender` and now cache queued transaction, we no longer have to cache the deeplink view.

## How to test it
- Create an out-of-order transaction and observe that there is no over-polling for the queued transactions occuring.
- Create a queued transaction, let it open in the deeplink view and navigate to the queue. Observe no duplicates or only the one deeplink-loaded transaction (if others are present).
- There should be no render issues or flickering occuring on the deeplink view.
